remove widget favicon coupling and move favicon code from core to the app

brindy opened the pull request:
Task/Issue URL: https://app.asana.com/0/414709148257752/1208785980508806/f
Tech Design URL:
CC:

**Description**:
Moves the favicon code the app bundle to make future refactoring easier.

Includes refactoring from #3593 

**Steps to test this PR**:
1. Run an old version of the app and add some favorites
3. Add the favorites widget and validate favicons appear
2. Upgrade to this version of the app
4. Open some tabs and ensure that the favicons appear in the tab switcher.
5. Check the cache location for the images. Look for `favicons tabs location` in Xcode console and go to the parent of that directory.
6. Also check the `favicons fireproof location`
7. Use the fire button.  
8. Ensure that images are removed from the tabs location but not from the fireproof location
9. Repeat with a clean install

brindy committed:
first step in decoupling widgets and favorites

brindy committed:
move majority of favicon code to app bundle

brindy committed:
move mock to own file and fix compile error

bwaresiak reviewed (approved):
Great job, it's much cleaner, caches work and clean as expected. 🎉

brindy committed:
rename preserved logins to fireproofing and use protocol where possible (#3593)

Task/Issue URL:
https://app.asana.com/0/392891325557410/1208796954308702/f
Tech Design URL:
CC: @bwaresiak 

**Description**:
Rename preserved logins to fireproofing and use protocol where possible

This is stacked on https://github.com/duckduckgo/iOS/pull/3590 - am
trying to make baby steps towards removing a few of the gnarlier
singletons (Favicons and Fireproofing at least)

**Steps to test this PR**:
1. Smoke test fireproofing with cookies, you can use
https://www.setcookie.net (remember to set an expiry date for it to
work)
2. Go into Settings > Data Clearing > Fireproof sites and make sure fire
proofed sites are visible and can be deleted from there
3. Check it works and visible in the settings UI
